src/core/echarts.ts

```diff
@@ -693,10 +685,30 @@ class ECharts extends Eventful<ECEventDefinition> {
     }
 
     /**
-     * @deprecated
+     * Update theme with name or theme option and repaint the chart.
+     * @param theme Theme name or theme option.
      */
-    private setTheme(): void {
-        deprecateLog('ECharts#setTheme() is DEPRECATED in ECharts 3.0');
+    setTheme(theme: string | ThemeOption): void {
+        this._updateTheme(theme);
+        if (this._model) {
+            this._model.setTheme(this._theme);
+            // Force refresh to apply theme changes
+            updateMethods.prepareAndUpdate.call(this, {
+                type: 'themeChanged'
```

Comment:
I think the `type: 'themeChanged'` is ineffective. It won't tigger an event. it is probably inappropriate to trigger an event in an explicit API call (which might lead to dead loop).
If needing to call `updateMethods.prepareAndUpdate` or `updateMethods.update`, it's OK to use `null` as the 2nd parameter.

Reply:
Do you mean `updateMethods.prepareAndUpdate.call(this);`? This causes `optionChanged: payload.newOption != null` to have error since `payload` is `undefined`. It works if I add `payload &&` as:

```ts
updateMethods.update.call(this, payload, payload && {
    // Needs to mark option changed if newOption is given.
    // It's from MagicType.
    // TODO If use a separate flag optionChanged in payload?
    optionChanged: payload.newOption != null
});
```

Reply:
I prefer to add a null checker in `prepareAndUpdate` 
or call 
```js
prepare(this);
updateMethods.update.call(this);
``` 
instead.

Do not use a payload because I think that a payload typically suggests this in an action but this is a explicit call rather than trigger an action.

Reply:
I changed my understanding, it's OK to use
```js
{
                type: 'themeChanged'
}
```
But there is more details need to be considered, see comments below.